Source/Core/Core/AchievementManager.cpp

```diff
@@ -1020,8 +1020,15 @@ void AchievementManager::FetchBadge(AchievementManager::Badge* badge, u32 badge_
 
     if (!LoadPNGTexture(badge, *http_response))
     {
-      ERROR_LOG_FMT(ACHIEVEMENTS, "Default game badge '{}' failed to load",
-                    DEFAULT_GAME_BADGE_FILENAME);
+      ERROR_LOG_FMT(ACHIEVEMENTS, "Badge '{}' failed to load", name_to_fetch);
+    }
+
+    std::string cache_base = File::GetUserPath(D_RETROACHIEVEMENTSCACHE_IDX);
+    std::string cache_path = fmt::format("{}/{}-{}.png", cache_base, badge_type, name_to_fetch);
```

Comment:
This needs the same filename logic as the loading.

Reply:
(please make it an actual function that both places call, so you don't have duplicated logic that may mismatch)

Reply:
Uh, look again, I only generate this path once in the current algorithm, and then use it for both the load and the save.

Reply:
Ah oops, I must have been looking at the old commit.

Reply:
Okay I'm looking at this again and I now realize what's happening here. It's because your first commit is committing this incorrectly and then the second commit fixes it.